lib/Dialect/LWE/IR/NewLWEAttributes.td

```diff
@@ -271,30 +271,27 @@ def LWE_PlaintextSpaceAttr : AttrDef<LWE_Dialect, "PlaintextSpace"> {
 def LWE_KeyAttr : AttrDef<LWE_Dialect, "Key"> {
   let mnemonic = "key";
   let description = [{
-    An attribute describing the key used for encrypting the ciphertext.
-
-    This attribute includes a key identifier for the original key used to
-    encrypt the secret key.
-
-    The `key_size` parameter is used to describe the number of polynomials of
-    the secret key. This is typically $1$ for RLWE ciphertexts and greater than
-    $1$ for LWE instances. A ciphertext encrypted with a `key_size` of $k$ will
-    have size $k+1$.
-
-    The key basis describes the inner product used in the phase calculation in
-    decryption. This attribute is only supported for RLWE ciphertexts whose
-    `key_size` is $1$. An RLWE ciphertext is canonically encrypted against key
-    basis `(1, s)`. After a multiplication, its size will increase and the basis
-    will be `(1, s, s^2)`. The array that represents the key basis is
-    constructed by listing the powers of `s` at each position of the array. For
-    example, `(1, s, s^2)` corresponds to `[0, 1, 2]`, while `(1, s^2)`
-    corresponds to `[0, 2]`.
+    An attribute describing the key used for decrypting the ciphertext.
+
+    The key attribute describes the key with which decryption is performed. For
+    example, if the decryption of a ciphertext $c = (c_0(x), c_1(x))$ is
+    performed by computing the inner product $(c_0(x), c_1(x)) \cdot (1, s(x))$
+    then the decryption key has a basis $(1, s(x))$.
+
+    The `rotation_basis` describes the powers of `x` at each position of the
+    array, so $(1, s(x))$ becomes $[0, 1]$. After a Galois rotation mapping $x
+    -> x^i$, the key will have a new basis $(1, s(x^i))$, which will be
+    represented by $[0, i]$.
+
+    Note that the ciphertext size must match the size of the key's basis.
+
+    By default, the rotation basis is assumed to be $[0, 1, ..., 1]$ so that
+    after a multiplication of two ciphertexts, the `rotation_basis` is $[0, 1,
+    1]$.
   }];
 
   let parameters = (ins
-    "::mlir::StringAttr":$id,
-    DefaultValuedParameter<"unsigned", "1">:$size,
-    OptionalArrayRefParameter<"unsigned int">:$basis
+    OptionalArrayRefParameter<"unsigned int">:$rotation_basis
```

Comment:
The description above makes it sound like it should be called "decryption_basis", but I am in favor of just calling it "basis" for brevity.